src/Sylius/Component/Core/spec/Statistics/Provider/OrdersTotals/DayBasedOrdersTotalProviderSpec.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Component\Core\Statistics\Provider\OrdersTotals;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Core\Statistics\Provider\OrdersTotals\OrdersTotalsProviderInterface;
+
+final class DayBasedOrdersTotalProviderSpec extends AbstractOrdersTotalsProviderSpec
+{
+    protected const DATE_FORMAT = 'Y-m-d';
+
+    private const GROUP_SELECT = [
+        'year' => 'YEAR(o.checkoutCompletedAt) AS year',
+        'month' => 'MONTH(o.checkoutCompletedAt) AS month',
+        'day' => 'DAY(o.checkoutCompletedAt) AS day',
+    ];
+
+    function let(OrderRepositoryInterface $orderRepository): void
+    {
+        $this->beConstructedWith($orderRepository);
+    }
+
+    function it_is_a_orders_totals_provider(): void
+    {
+        $this->shouldImplement(OrdersTotalsProviderInterface::class);
+    }
+
+    function it_returns_an_array_of_zeros_when_no_totals_have_been_found_for_period(
```

Comment:
Maybe something like `it returns zero as total when no totals have been found` or something like that. There is a single zero.

Reply:
Added more zeros 💃